src/main/cpp/util/md5.cc

```diff
@@ -206,10 +207,15 @@ void Md5Digest::Transform(
   // ROTATE_LEFT rotates x left n bits.
 #define ROTATE_LEFT(x, n) (((x) << (n)) | ((x) >> (32-(n))))
 
+// SET reads 4 input bytes in little-endian byte order and stores them
+// in a properly aligned word in host byte order.
+#define SET(n) \
```

Comment:
This could be a much smaller change.  I think it works equally well

- without the SET macro.
- the FF signature remains unchnaged
- (original) line 212 becomes
```
  (a) += F((b), (c), (d)) + ((*x_pos++ = h32tole(*cur_word++))) + \
```
Or, for safety against h32tole being a strange macro
```
  (a) += F((b), (c), (d)) + ((*x_pos++ = h32tole(*cur_word))) + \
     ...
  cur_word++;
```

And, doesn't this have to be h32tole()? We are pulling from host buffer to a little endian crypt engine.

Reply:
@aiuto Yes, this will be a better solution. I've tested your code change in #17966 and it worked fine. Thanks for your improvement!